tests/observability/suite.go

```diff
@@ -271,3 +279,104 @@ func testPodDisruptionBudgets(check *checksdb.Check, env *provider.TestEnvironme
 
 	check.SetResult(compliantObjects, nonCompliantObjects)
 }
+
+// Convert the list of APIRequestCount into a list of ToBeRemovedWorkloadAPI
+// using the following filters:
+// - status.removedInRelease is not nil
+// - service accounts in status.last24.byNode.ByUser.username do not contain the 'openshift' or 'kube' regex
+func buildToBeRemovedWorkloadAPIList(apiRequestCounts []provider.APIRequestCount) []provider.ToBeRemovedWorkloadAPI {
+	var toBeRemovedWorkloadAPIs []provider.ToBeRemovedWorkloadAPI
+
+	for _, obj := range apiRequestCounts {
+		if obj.Status.RemovedInRelease != "" {
+			workloadAPI := provider.ToBeRemovedWorkloadAPI{
+				APIName:          obj.Metadata.Name,
+				RemovedInRelease: obj.Status.RemovedInRelease,
+				UserInfo:         make(map[string]struct{}),
+			}
+
+			for _, last24h := range obj.Status.Last24h {
+				for _, byNode := range last24h.ByNode {
+					if byNode.ByUser != nil {
+						for _, byUser := range byNode.ByUser {
+							if !strings.Contains(byUser.UserName, "openshift") && !strings.Contains(byUser.UserName, "kube") {
+								userInfo := fmt.Sprintf("UserName: %s, UserAgent: %s", byUser.UserName, byUser.UserAgent)
+								workloadAPI.UserInfo[userInfo] = struct{}{}
+							}
+						}
+					}
+				}
+			}
+
+			if len(workloadAPI.UserInfo) > 0 {
+				toBeRemovedWorkloadAPIs = append(toBeRemovedWorkloadAPIs, workloadAPI)
+			}
+		}
+	}
+
+	return toBeRemovedWorkloadAPIs
+}
+
+// Evaluate workload API compliance with the next Kubernetes version
+func evaluateAPICompliance(toBeRemovedWorkloadAPIs []provider.ToBeRemovedWorkloadAPI, kubernetesVersion string) (compliantObjects, nonCompliantObjects []*testhelper.ReportObject) {
+	// Extract the major version from the current Kubernetes version
+	minorVersion, err := strconv.Atoi(strings.Split(kubernetesVersion, ".")[1])
+	if err != nil {
+		fmt.Printf("Error parsing minor version from Kubernetes version: %s\n", err)
+		return nil, nil
+	}
+
+	for _, api := range toBeRemovedWorkloadAPIs {
+		removedMinorVersion, err := strconv.Atoi(strings.Split(api.RemovedInRelease, ".")[1])
+		if err != nil {
+			fmt.Printf("Error parsing minor version from APIRequestCount.status.removedInRelease: %s\n", err)
+			// Skip this API if the version parsing fails
+			continue
+		}
+
+		isCompliantWithNextK8sVersion := removedMinorVersion > minorVersion+1
+		reportObject := testhelper.NewReportObject("API", "StubType", isCompliantWithNextK8sVersion)
```

Comment:
good point, thank you, updated the test results to look like this
```
"checkDetails": "{\"CompliantObjectsOut\":null,
\"NonCompliantObjectsOut\":
[{\"ObjectType\":\"API\",\"ObjectFieldsKeys\":
[\"Reason For Non Compliance\",
\"RemovedInRelease\",\"APIName\",\"UserInfo\",
\"UserInfo\"],\"ObjectFieldsValues\":
[\"API is not compliant with Kubernetes version 1.25.0 
because it will be removed in release 1.25\",\"1.25\",
\"events.v1beta1.events.k8s.io\",
\"UserName: system:serviceaccount:default:eventtest-operator-service-account, 
UserAgent: OpenAPI-Generator/24.2.0/python\",
\"UserName: system:serviceaccount:default:eventtest-operator-service-account, 
UserAgent: ansible-operator/v0.0.0\"]}]}",
```